src/Microsoft.Identity.Client/PublicClientApplication.WithBuilders.cs

```diff
@@ -171,13 +174,26 @@ internal PublicClientApplication(ApplicationConfiguration configuration)
             requestParams.LoginHint = interactiveParameters.LoginHint;
             requestParams.Account = interactiveParameters.Account;
 
-            var handler = new InteractiveRequest(
+            if (requestParams.IsBrokerEnabled)
+            {
+                AcquireTokenByBrokerParameters brokerParameters = new AcquireTokenByBrokerParameters();
```

Comment:
I like that you've created a separate request, definitely cleaner than ADAL. Not much point in creating an empty dictionary for the params though...